src/core/api/clock.ts

```diff
@@ -349,14 +351,22 @@ function createClock(
   return observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime,
+        mediaInternalSeeking: options.currentTimeHandle.isSeekingFromInside,
+      });
 
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
       const mediaTimings = getMediaInfos(mediaElement, state);
       const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
+      if (options.currentTimeHandle.isSeekingFromInside && stalledState === null) {
+        options.currentTimeHandle.isSeekingFromInside = false;
```

Comment:
I don't follow how `isSeekingFromInside` is used, despite just being switched from `true` to `false`.
Am I missing something or is it just the first step (and the next step being: don't send seek related events in those cases)?

For that second step, we have to avoid at least:

  1. A `playerStateChange` event to `"SEEKING"`.
            `getPlayerState.ts` rely on the clock, more specifically on `stalled.reason` to know when that event has to be sent.
            We could thus just be adding a category to `stalled.reason`, like "internal-seek" - this is only a suggestion and may not be the best way to do it.

  2. A `seeking` and a `seeked` event. This might be just as easy/hard than for `playerStateChange` because the clock is also used in `emit_seek_events.ts` which decides when to send those.
                Here we just might: 
                   1. Just rely on the clock's `state` property and not on media element events to emit "seeking" and "seeked" events.
                   2. Find some way to signal here that this is only an internal-seek (e.g. also setting `state` to a `"internal-seek"` value? We could also add another property or rely on already-existing ones here).

Reply:
Done